Backend/nix/osrm.nix

```diff
@@ -3,7 +3,7 @@
 let
   openStreetDataFile = inputs.osrm-pbf;
   # NOTE: This *should* match the flake input.
-  openStreetDataFileName = "southern-zone-231218";
+  openStreetDataFileName = "southern-zone-231222";
```

Comment:
Let's change this to use Jan 1

cf. https://github.com/nammayatri/nammayatri/pull/4783